code/modules/mob/living/simple_animal/hostile/regalrat.dm

```diff
@@ -0,0 +1,214 @@
+/mob/living/simple_animal/hostile/regalrat
+	name = "regal rat"
+	desc = "An evolved rat, created through some strange science. It leads nearby rats with deadly efficiency to protect its kingdom. Not technically a king."
+	icon_state = "regalrat"
+	icon_living = "regalrat"
+	icon_dead = "regalrat_dead"
+	gender = NEUTER
+	speak_chance = 0
+	turns_per_move = 5
+	maxHealth = 70
+	health = 70
+	see_in_dark = 5
+	obj_damage = 10
+	butcher_results = list(/obj/item/clothing/head/crown = 1,)
+	response_help_continuous = "glares at"
+	response_help_simple = "glare at"
+	response_disarm_continuous = "skoffs at"
+	response_disarm_simple = "skoff at"
+	response_harm_continuous = "slashes"
+	response_harm_simple = "slash"
+	melee_damage_lower = 13
+	melee_damage_upper = 15
+	attack_verb_continuous = "slashes"
+	attack_verb_simple = "slash"
+	attack_sound = 'sound/weapons/punch1.ogg'
+	ventcrawler = VENTCRAWLER_ALWAYS
+	faction = list("rat")
+	var/datum/action/cooldown/coffer
+	var/datum/action/cooldown/riot
+	///Number assigned to rats and mice, checked when determining infighting.
+	var/faction_num = 0
+
+/mob/living/simple_animal/hostile/regalrat/Initialize()
+	. = ..()
+	coffer = new /datum/action/cooldown/coffer
+	riot = new /datum/action/cooldown/riot
+	coffer.Grant(src)
+	riot.Grant(src)
+	faction_num = rand(1,999)
```

Comment:
there is a `tag` var on all mobs (unique) that you can add to other mobs' factions rather than making a completely new var that does what faction would already handle.